d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -452,4 +576,17 @@ private int getNumberOfPartitions()
     Map<Integer, PartitionData> partitionDataMap = getPartitionData();
     return partitionDataMap == null ? 0 : partitionDataMap.size();
   }
+
+  /**
+   * A way to force users to set both executor service and warm up parameters, if they want to use d2 warmup
+   */
+  public void setExecutorServiceAndWarmupParameters(ScheduledExecutorService executorService, boolean isEnabledD2WarmupCluster,
+      String warmupClusterName, int warmupDuration)
+  {
+    _executorService = executorService;
+    _isEnabledD2WarmupCluster = isEnabledD2WarmupCluster;
+    _warmupClusterName = warmupClusterName;
+    _warmupDuration = warmupDuration;
+  }
```

Comment:
What are you seeing in the container code that makes this necessary (rather than just using the new constructor)?

Reply:
Thanks for noticing this! In the container code, currently ZooKeeperAnnouncer object is by using the second constructor
ZooKeeperAnnouncer(ZooKeeperServer server, boolean initialIsUp)

and setter methods are used for other class parameters like _cluster, _uri, _partitionDataMap and _uriSpecificProperties.

Reply:
Should I pasd just the warm up parameters through the new constructor? or others  like _cluster, _uri, _partitionDataMap and _uriSpecificProperties. too?

Reply:
Deleted this method as for warm-up parameters only constructor should be used, as these should not change at runtime.